src/runtime_src/core/edge/user/aie/graph.cpp

```diff
@@ -230,26 +194,26 @@ wait_done(int timeout_ms)
      */
     while (1) {
         uint8_t done;
-        for (auto& tile : tiles) {
+        for (int i = 0; i < graph_config.coreColumns.size(); i++){
             /* Skip multi-rate core */
-            if (tile.is_trigger) {
+            if (graph_config.triggered[i]) {
                 done = 1;
                 continue;
             }
 
-            XAie_LocType coreTile = XAie_TileLoc(tile.col, tile.row);
+            XAie_LocType coreTile = XAie_TileLoc(graph_config.coreColumns[i], graph_config.coreRows[i] + adf::config_manager::s_num_reserved_rows + 1);
```

Comment:
Just curious, why can't we use a tile class rather than use separate column and row vector? The code at left side looks much cleaner to me :)

Reply:
Will address it in the follow up pull request